x-pack/filebeat/input/azureeventhub/input.go

```diff
@@ -126,6 +178,11 @@ func (a *azureInput) Stop() {
 		}
 	}
 
+	if a.metrics != nil {
+		a.log.Debugf("%s deregistering input metrics for input with ID %s", inputName, a.id)
+		a.metrics.Close()
+	}
```

Comment:
Prefer placing this guard in `*inputMetrics.Close` without logging. Note that [metric registration and deregistration are logged in inputmon with an orthogonal ID](https://github.com/elastic/beats/blob/3ae95ff85756c3ca71fff4d926814daf62a026c0/libbeat/monitoring/inputmon/input.go#L55-L71), so that doesn't need to be done here.

Reply:
Logging: you're right! Removing this one.